recipes/physx/4.1.1/conanfile.py

```diff
@@ -0,0 +1,277 @@
+
+        # Options defined in physx/compiler/public/CMakeLists.txt
+        cmake.definitions["PX_BUILDSNIPPETS"] = False
+        cmake.definitions["PX_BUILDPUBLICSAMPLES"] = False
+        cmake.definitions["PX_CMAKE_SUPPRESS_REGENERATION"] = False
+        cmakemodules_path = os.path.join(self._source_subfolder, "externals")
+        if self.settings.os == "Windows":
+            cmakemodules_path = os.path.join(cmakemodules_path, "CMakeModules")
+        else:
+            cmakemodules_path = os.path.join(cmakemodules_path, "cmakemodules")
+        cmake.definitions["CMAKEMODULES_PATH"] = os.path.abspath(cmakemodules_path).replace("\\", "/")
+        cmake.definitions["PHYSX_ROOT_DIR"] = os.path.abspath(os.path.join(self._source_subfolder, "physx")).replace("\\", "/")
+
+        # Options defined in physx/source/compiler/cmake/CMakeLists.txt
+        cmake.definitions["PX_SCALAR_MATH"] = not self.options.enable_simd
+        cmake.definitions["PX_GENERATE_STATIC_LIBRARIES"] = not self.options.shared
+        cmake.definitions["PX_EXPORT_LOWLEVEL_PDB"] = False
+        cmake.definitions["PXSHARED_PATH"] = os.path.abspath(os.path.join(self._source_subfolder, "pxshared")).replace("\\", "/")
+        cmake.definitions["PXSHARED_INSTALL_PREFIX"] = self.package_folder.replace("\\", "/")
+
+        # Options defined in externals/cmakemodules/NVidiaBuildOptions.cmake
+        cmake.definitions["NV_APPEND_CONFIG_NAME"] = False
+        cmake.definitions["NV_USE_GAMEWORKS_OUTPUT_DIRS"] = False
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.compiler.runtime == "MT":
+                cmake.definitions["NV_USE_STATIC_WINCRT"] = True
+                cmake.definitions["NV_USE_DEBUG_WINCRT"] = False
+            elif self.settings.compiler.runtime == "MTd":
+                cmake.definitions["NV_USE_STATIC_WINCRT"] = True
+                cmake.definitions["NV_USE_DEBUG_WINCRT"] = True
+            else:
+                cmake.definitions["NV_USE_STATIC_WINCRT"] = False
+                cmake.definitions["NV_USE_DEBUG_WINCRT"] = False
+        else:
+            cmake.definitions["NV_USE_STATIC_WINCRT"] = False
+            cmake.definitions["NV_USE_DEBUG_WINCRT"] = False
+        cmake.definitions["NV_FORCE_64BIT_SUFFIX"] = False
+        cmake.definitions["NV_FORCE_32BIT_SUFFIX"] = False
+        cmake.definitions["PX_ROOT_LIB_DIR"] = os.path.join(self.package_folder, "lib").replace("\\", "/")
+
+        # Options defined in physx/source/compiler/cmake/windows/CMakeLists.txt
+        if self.settings.os == "Windows":
+            cmake.definitions["PX_COPY_EXTERNAL_DLL"] = False # External dll copy disabled, PhysXDevice dll copy is handled afterward during conan packaging
+            cmake.definitions["PX_FLOAT_POINT_PRECISE_MATH"] = self.options.enable_float_point_precise_math
+            cmake.definitions["PX_USE_NVTX"] = False # Could be controlled by an option if NVTX had a recipe, disabled for the moment
```

Comment:
Maybe it would be a good idea to also add `cmake.definitions["PX_GENERATE_GPU_PROJECTS"] = False` to clearly state that its a cpy-only Physx.

Reply:
I agree, explicit is better than implicit